### ISSUE TITLE
unable to run deepvariant using conda

### ISSUE THREAD

**Issue by Zer0day-0** [google/deepvariant#806]
**Describe the issue:**
while trying to install deepvariant with conda its is running for dour days, still nothing is getting installed.

**Setup**
  - OS: CentOS Linux release 7.4.1708 (Core)
  - DeepVariant version:conda install bioconda/label/cf201901::deepvariant
  - Installation method (Docker, built from source, etc.): Conda
  - Type of data: NA

**Steps to reproduce:**
  - Command: conda install bioconda/label/cf201901::deepvariant
  - Error trace: 
'''conda install bioconda/label/cf201901::deepvariant -y
Collecting package metadata: done
Solving environment: '''

**Does the quick start test work on your system?**
Please test with https://github.com/google/deepvariant/blob/r1.6/docs/deepvariant-quick-start.md.
Is there any way to reproduce the issue by using the quick start? NA

**Any additional context:**
NA

**Comment by pichuan** [comment]
Hi @Zer0day-0 ,
thanks for the report. DeepVariant conda actually isn't maintained by our team. I can take a look later this week. 

But before that , I have a question - is it possible for you to use other solutions, such as our Docker version or Singularity?
If you can describe your situation and why you might only be able to use conda, that will also be useful context for us. 

Thank you!

**Comment by Zer0day-0** [comment]
Hello @pichuan, 
Thanks for response, actually I am running a SGE cluster in order to run my analysis, and the sysadmins keeps use of docker restricted(root access issue). Hence I am tryng to use conda.

**Comment by pichuan** [comment]
Thank you. Would you be able to use Singularity?

**Comment by Zer0day-0** [comment]
Nope, similar issue of sysadmin!

**Comment by pichuan** [comment]
Thank you @Zer0day-0 .

I went back and looked at my notes on the last time I tried to install with Conda: https://github.com/google/deepvariant/issues/736#issuecomment-1829204521

I noticed that you're using CentOS. I think I might be testing with CentOS at the time. So let me try that.

# Get a CentOS machine to test.

I used:

```bash
gcloud compute instances create "${USER}-centos7" \
--scopes "compute-rw,storage-full,cloud-platform" \
--image-family "centos-7" \
--image-project "centos-cloud" \
--machine-type "e2-standard-16" \
--zone "us-west1-b"

gcloud compute ssh pichuan-centos7 --zone "us-west1-b"
```

Version:

```
$ uname -a
Linux pichuan-centos7 3.10.0-1160.114.2.el7.x86_64 #1 SMP Wed Mar 20 15:54:52 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

Install conda:

```bash
curl https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh > Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh -b -u -p $HOME/miniconda
eval "$(${HOME}/miniconda/bin/conda shell.bash hook)"
```

To repeat what I did in, I tried: https://github.com/google/deepvariant/issues/736#issuecomment-1829204521

```bash
conda config --add channels defaults && \
conda config --add channels bioconda && \
conda config --add channels conda-forge
conda create -y -n dv-env deepvariant
conda activate dv-env
```

It completed without any error messages. I see:

```
(dv-env) [pichuan@pichuan-centos7 ~]$ ls /home/pichuan/miniconda/envs/dv-env/share/
bash-completion  deepvariant-1.5.0-0  doc  et  examples  google-cloud-sdk-359.0.0-0  icu  info  keyutils  licenses  locale  man  tabset  terminfo  zsh
(dv-env) [pichuan@pichuan-centos7 ~]$ ls /home/pichuan/miniconda/envs/dv-env/share/deepvariant-1.5.0-0/binaries/DeepVariant/1.5.0/DeepVariant-1.5.0
call_variants_keras.zip  freeze_graph.zip           make_examples.zip  multisample_make_examples.zip  runtime_by_region_vis.zip  vcf_stats_report.zip
call_variants.zip        licenses.zip               model_eval.zip     postprocess_variants.zip       settings.sh
deeptrio                 make_examples_somatic.zip  model_train.zip    run-prereq.sh                  show_examples.zip
```

As mentioned in my previous investigation, I don't actually know how to use Conda though. But it seems like at least with the setting above, it can install.

If you can try that and see if it works for you. And do what you'd usually do to run and see if it works, that will be great! Please let me know.


By the way, 
I noticed you were using `bioconda/label/cf201901::deepvariant`. I actually don't know what that does. (Can you explain where that came from and why you needed that?)
But to understand be behavior, I also tried something like:
`conda create -y -n dv-env bioconda/label/cf201901::deepvariant`

If I did that, then it still seems to be work, but it somehow had an older version `deepvariant-0.7.2-1` in /home/pichuan/miniconda/envs/dv-env/share/, which I don't understand why.

I'd recommend that you run using my first command, so you can have a more recent version to try out!

**Comment by Zer0day-0** [comment]
thank you for your kind support @pichuan 

It seems I was running a really old and obsolete version of anaconda(4.2.1) in my SGE. It took some gymnastics to install it for a single user and run across the nodes but I updated my conda to current version. after that installation completed with no errors. 

the "bioconda/label/cf201901::deepvariant" was provided in bioconda page, which is supposed to be another installation solution. I think this issue can be closed. For anyone in future running in a problem similar to me, I'll make a repo with the installation code in my profile!

**Comment by Zer0day-0** [comment]
Hi @jankinsan Please check out [here](https://github.com/Zer0day-0/SGE_nightmare_solve/blob/d77dae0c34038c3a3cc97e3b690adada3f7c5dbb/conda_user_specific_install.sh) for the solution, just out of curiosity, are you using a SGE or SLURM cluster ?
